flink-core/src/main/java/org/apache/flink/core/fs/FileSystem.java

```diff
@@ -588,8 +588,10 @@ public static URI getDefaultFsUri() {
      *
      * @param f The path we want information from
      * @return a FileStatus object
-     * @throws FileNotFoundException when the path does not exist; IOException see specific
-     *     implementation
+     * @throws FileNotFoundException when the path does not exist. This also applies to cases where
+     *     the {@code FileSystem} implementation is based on an object store like S3 and the passed
+     *     {@code Path} refers to a directory (even if it was created through {@link #mkdirs(Path)};
```

Comment:
Looking at the S3 FileSystem implementation I'm not sure if this is generally correct. Directories _technically_ don't exist; when you do a mkdirs it creates an object that pretends to be a directory (in the plain hadoop s3 filesystem).

Reply:
You're right. I reverted the JavaDoc changes entirely because I feel that it adds more confusion than it helps. FLINK-26061 is still valid, though.